src/gluonnlp/data/transforms.py

```diff
@@ -988,10 +1004,261 @@ def convert_tokens_to_ids(self, tokens):
+            (text_a, text_b). For single sequences, the input is a tuple of single
+            string: (text_a,).
+
+        Returns
+        -------
+        np.array: input token ids in 'int32', shape (batch_size, seq_length)
+        np.array: valid length in 'int32', shape (batch_size,)
+        np.array: input token type ids in 'int32', shape (batch_size, seq_length)
+        """
+        # convert to unicode
+        text_a = line[0]
+        if self._pair:
+            assert len(line) == 2
+            text_b = line[1]
+
+        tokens_a = self._tokenizer(text_a)
+        tokens_b = None
+
+        if self._pair:
+            tokens_b = self._tokenizer(text_b)
+
+        if tokens_b:
+            # Modifies `tokens_a` and `tokens_b` in place so that the total
+            # length is less than the specified length.
+            # Account for [CLS], [SEP], [SEP] with "- 3"
+            _truncate_seq_pair(tokens_a, tokens_b, self._max_seq_length - 3)
+        else:
+            # Account for [CLS] and [SEP] with "- 2"
+            if len(tokens_a) > self._max_seq_length - 2:
+                tokens_a = tokens_a[0:(self._max_seq_length - 2)]
+
+        # The embedding vectors for `type=0` and `type=1` were learned during
+        # pre-training and are added to the wordpiece embedding vector
+        # (and position vector). This is not *strictly* necessary since
+        # the [SEP] token unambiguously separates the sequences, but it makes
+        # it easier for the model to learn the concept of sequences.
+
+        # For classification tasks, the first vector (corresponding to [CLS]) is
+        # used as as the "sentence vector". Note that this only makes sense because
+        # the entire model is fine-tuned.
+        tokens = []
+        segment_ids = []
+        tokens.append('[CLS]')
+        segment_ids.append(0)
+        for token in tokens_a:
+            tokens.append(token)
+            segment_ids.append(0)
+        tokens.append('[SEP]')
+        segment_ids.append(0)
+
+        if tokens_b:
+            for token in tokens_b:
+                tokens.append(token)
+                segment_ids.append(1)
+            tokens.append('[SEP]')
+            segment_ids.append(1)
+
+        input_ids = self._tokenizer.convert_tokens_to_ids(tokens)
+
+        # The valid length of sentences. Only real  tokens are attended to.
+        valid_length = len(input_ids)
+
+        if self._pad:
+            # Zero-pad up to the sequence length.
+            padding_length = self._max_seq_length - valid_length
+            # use padding tokens for the rest
+            input_ids.extend([self._tokenizer.vocab['[PAD]']] * padding_length)
+            segment_ids.extend([self._tokenizer.vocab['[PAD]']] * padding_length)
+
+        return np.array(input_ids, dtype='int32'), np.array(valid_length, dtype='int32'),\
+            np.array(segment_ids, dtype='int32')
+
+
+class BERTClassificationTransform(object):
```

Comment:
Can we unify the interface for classification and regression? We can have a reusable transform for both classification and regression. 
The transform should take a label_dtype argument (default to 'float32'). For classification users can set dtype to 'int32' explicitly.

Reply:
Ok, this is easy to modify.